lessons/5.mdx

```diff
@@ -0,0 +1,1362 @@
+}
+```
+
+</ ContentSideDrawer>
+<br/>
+
+Moving onto imports.
+
+```jsx
+// prev imports...
+
+import { ConnectButton } from '@rainbow-me/rainbowkit';
+import { useAccount, useContractRead, useContractWrite } from 'wagmi';
+import TierABI from '../../artifacts/contracts/TierNFT.sol/TierNFT.json';
+import styles from '../styles/Home.module.css';
+import { ethers } from 'ethers';
+import { useEffect, useState } from 'react';
+
+// function Home code...
+```
+
+- RainbowKit provides us a ready-made `ConnectButton`.
+- We need to the `useAccount` hook from WAGMI to check account connection 
+and `useContractRead` and `useContractWrite` to interact with our smart 
+contract. We will look into WAGMI and these hooks a little down the line.
+- `TierABI` is our contract's Application Binary Interface (ABI). The ABI 
+helps us interact with the smart contract outside the blockchain or 
+contract-to-contract. You can read more about ABIs here.
+- The `ethers` library will assist us with some utilities in our code.
+- Once again, `useEffect` and `useState` are react hooks that will help us 
+perform side effects and capture the state, respectively.
+
+```jsx
+export default function Home() {
+  const CONTRACT_ADDRESS = /*Enter Contract Address between ''*/;
+
+  const { isConnected } = useAccount();
+
+  const [isUserConnected, setIsUserConnected] = useState(false);
+  const [latestNFTMinted, setLatestNFTMinted] = useState(0);
+  const [modalShow, setModalShow] = useState(false);
+  const [isMinting, setIsMinting] = useState(false);
+
+  // More code to come...
+}
+```
+
+We're creating a default function called `Home` that will render our home/
+index page.
+The `useAccount` hook lets us access account data and check connection status.
+We will use this as a check to provide access to the minting features of our
+app.
+A few `useState`s let use store state information. The general syntax for 
+using the `useState` hook is:
+
+`const [variableName, functionToSetVariable] = useState(intialValue)`
+
+<ContentCallout emoji="💡">
+  The initial value also serves as the type for the value assigned. If the 
+  initial value is a string, then the values set later must be strings too.
+  If value type is not known, we can choose to not pass in any initalValue,
+  however, it is not good practice to do so. 
+</ContentCallout>
+
+We will revisit these state variables later to see their implementation.
+
+### Header
+
+We need a `<header>` that displays our application name and let's us 
+connect our wallet.
+
+```jsx
+export default function Home() {
+  // variable definitions...
+
+  return (
+      <div className={styles.container}>
+        
+        {/* Head tag... */}
+
+        <header style={header}>
+          <h1>TierNFTs</h1>
+          <ConnectButton />
+        </header>
+
+        {/* More code... */}
+
+      </div>
+  )
+}
+
+// styling code...
+```
+
+For accesibility we must ensure we are using only a single `h1` tag on 
```

Comment:
would you consider the following?
Since we care about accessibility, we ensure we are using only a single `<h1>` element on a page and the correct sub-heading elements as well.
Starting the sentence with **For accessibility.....** is a bit lonely and therefore vague